cumulus/polkadot-parachain/src/chain_spec/coretime.rs

```diff
@@ -0,0 +1,285 @@
+	pub const ID_PREFIX: &'static str = "coretime";
+
+	pub fn load_config(&self) -> Result<Box<dyn ChainSpec>, String> {
+		match self {
+			CoretimeRuntimeType::Rococo => Ok(Box::new(GenericChainSpec::from_json_bytes(
+				&include_bytes!("../../../parachains/chain-specs/coretime-rococo.json")[..],
+			)?)),
+			CoretimeRuntimeType::RococoLocal => Ok(Box::new(rococo::local_config(
+				rococo::CORETIME_ROCOCO_DEVELOPMENT,
+				"Rococo Coretime Local",
+				"rococo-local",
+				ParaId::new(1005),
+			))),
+			CoretimeRuntimeType::RococoDevelopment => Ok(Box::new(rococo::local_config(
+				rococo::CORETIME_ROCOCO_DEVELOPMENT,
+				"Rococo Coretime Development",
+				"rococo-dev",
+				ParaId::new(1005),
+			))),
+			CoretimeRuntimeType::Westend => Ok(Box::new(GenericChainSpec::from_json_bytes(
+				&include_bytes!("../../../parachains/chain-specs/coretime-westend.json")[..],
+			)?)),
+			CoretimeRuntimeType::WestendLocal => Ok(Box::new(westend::local_config(
+				westend::CORETIME_WESTEND_DEVELOPMENT,
+				"Westend Coretime Local",
+				"westend-local",
+				ParaId::new(1005),
+			))),
+			CoretimeRuntimeType::WestendDevelopment => Ok(Box::new(westend::local_config(
+				westend::CORETIME_WESTEND_DEVELOPMENT,
+				"Westend Coretime Development",
+				"westend-dev",
+				ParaId::new(1005),
+			))),
+		}
+	}
+}
+
+/// Check if `id` satisfies Coretime-like format.
+fn ensure_id(id: &str) -> Result<&str, String> {
+	if id.starts_with(CoretimeRuntimeType::ID_PREFIX) {
+		Ok(id)
+	} else {
+		Err(format!(
+			"Invalid 'id' attribute ({}), should start with prefix: {}",
+			id,
+			CoretimeRuntimeType::ID_PREFIX
+		))
+	}
+}
+
+/// Sub-module for Rococo setup.
+pub mod rococo {
+	use super::{GenericChainSpec, ParaId};
+	use crate::chain_spec::{
+		get_account_id_from_seed, get_collator_keys_from_seed, Extensions, SAFE_XCM_VERSION,
+	};
+	use parachains_common::{AccountId, AuraId, Balance};
+	use sc_chain_spec::ChainType;
+	use sp_core::sr25519;
+
+	pub(crate) const CORETIME_ROCOCO: &str = "coretime-rococo";
+	pub(crate) const CORETIME_ROCOCO_LOCAL: &str = "coretime-rococo-local";
+	pub(crate) const CORETIME_ROCOCO_DEVELOPMENT: &str = "coretime-rococo-dev";
+	const CORETIME_ROCOCO_ED: Balance = parachains_common::rococo::currency::EXISTENTIAL_DEPOSIT;
+
+	pub fn local_config(
+		id: &str,
+		chain_name: &str,
+		relay_chain: &str,
+		para_id: ParaId,
+	) -> GenericChainSpec {
+		// Rococo defaults
+		let mut properties = sc_chain_spec::Properties::new();
+		properties.insert("ss58Format".into(), 42.into());
+		properties.insert("tokenSymbol".into(), "ROC".into());
+		properties.insert("tokenDecimals".into(), 12.into());
+
+		GenericChainSpec::builder(
+			coretime_rococo_runtime::WASM_BINARY
+				.expect("WASM binary was not built, please build it!"),
+			Extensions { relay_chain: relay_chain.to_string(), para_id: para_id.into() },
+		)
+		.with_name(chain_name)
+		.with_id(super::ensure_id(id).expect("invalid id"))
+		.with_chain_type(ChainType::Local)
```

Comment:
Should this be `ChainType::Development` if we passed in a `-dev` chain?

Reply:
This is just a shortcut since the dev chain is otherwise identical to the local, but I can easily pass that as an argument instead.